### PR TITLE
Support Flink 1.16

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Close https://github.com/apache/incubator-kyuubi/issues/3513

https://www.mail-archive.com/dev@flink.apache.org/msg61009.html

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/OperationUtils.scala [hunk 1]
```diff
@@ -157,7 +162,11 @@ object OperationUtils {
       showJarsOperation: ShowJarsOperation,
       executor: Executor,
       sessionId: String): ResultSet = {
-    val jars = executor.listJars(sessionId)
+    // Removed by FLINK-27790
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/OperationUtils.scala:165]: Luckily, FLINK-27790 is the only breaking change for Kyuubi use cases

**Comment by pan3793** [comment]
cc @yanghua @SteNicholas @link3280

**Comment by yanghua** [comment]
> cc @yanghua @SteNicholas @link3280

+1 for supporting 1.16, can we find a good way to keep 1.14 for a longer time?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -146,15 +146,15 @@ jobs:
...
           - '1.15'
+          - '1.16'
         flink-archive: [ "" ]
         comment: [ "normal" ]
         include:
           - java: 8
             flink: '1.15'
-            flink-archive: '-Dflink.archive.mirror=https://archive.apache.org/dist/flink/flink-1.14.5 -Dflink.archive.name=flink-1.14.5-bin-scala_2.12.tgz'
-            comment: 'verify-on-flink-1.14-binary'
+            flink-archive: '-Dflink.archive.mirror=https://dist.apache.org/repos/dist/dev/flink/flink-1.16.0-rc0 -Dflink.archive.name=flink-1.16.0-bin-scala_2.12.tgz'
```
Comment (link3280) [inline .github/workflows/master.yml:156]: @pan3793 Are there any reasons we change to verifying forward compatibilities? I think the test is originally verifying backward compatibilities.

**Comment by link3280** [comment]
> > cc @yanghua @SteNicholas @link3280
> 
> +1 for supporting 1.16, can we find a good way to keep 1.14 for a longer time?

+1 to keep Flink 1.14 longer, as Flink 1.15 is released only 4 months ago (2022.05), and we have little overhead to maintain 1.14 support.

**Comment by yaooqinn** [comment]
+1 to keep 1.14,

**Comment by pan3793** [comment]
As all guys want to keep support for Flink 1.14, I reverted those parts of the changes.

**Comment by pan3793** [comment]
Thanks all for review, merging to master

**Pull request closed without merging**